Add internal server component

emdv88 opened the pull request.

emdv88 committed:
initial files copied

emdv88 committed:
remove unnecessary dependencies

emdv88 committed:
fix imports

emdv88 committed:
fix quality

emdv88 committed:
add internal-server to workflows

emdv88 committed:
internal dockerfile does not yet exist

fniessink commented on components/internal_server/README.md:
```diff
@@ -0,0 +1,3 @@
+# *Quality-time* external server
```
```suggestion
# *Quality-time* internal server
```

fniessink commented on components/internal_server/requirements/requirements-dev.in:
```diff
@@ -0,0 +1,3 @@
+-r ../../../requirements/requirements-dev.in
+-r ../../../requirements/types-requests.in
+types-cryptography==3.3.15
```
Why does the internal server need types-cryptography?

emdv88 replied:
it doesn't, good point

fniessink commented on components/internal_server/requirements/requirements.in:
```diff
@@ -0,0 +1,4 @@
+-r ../../../requirements/bottle.in
+-r ../../../requirements/lxml.in
+gevent==21.12.0
+pymongo==4.1.1
```
The external server needs gevent and pymongo too, maybe put them in ../../../requirements/gevent.in and ../../../requirements/pymongo.in, respectively?

emdv88 replied:
You use these requirements files to ensure identical versions?

fniessink replied:
Yes, and it's an attempt to reduce duplication among the requirements files.

fniessink commented on components/internal_server/setup.cfg:
```diff
@@ -0,0 +1,35 @@
+ignore_missing_imports = true
+
+[mypy-lxml.html.clean]
+ignore_missing_imports = true
```
Does the internal server use lxml?

emdv88 replied:
I don't know, it is still in the requirements. But I wouldn't know how to check 100% certain if we still need it.

emdv88 committed:
Update components/internal_server/README.md

Co-authored-by: Frank Niessink <frank@niessink.com>

emdv88 committed:
remove unused dev requirements

remove lxml dependency and see what happends

use shared requirements files for gevent and pymongo

pip compile

add pydantic

emdv88 committed:
update sonarcloud workflow

emdv88 committed:
temporarily disable pip-audit

emdv88 committed:
move shared_python dependencies to setup.cfg

emdv88 committed:
no -r

emdv88 committed:
safety check doesn't need requirements.txt anymore

emdv88 committed:
deadpendency shouldn't analyse shared python dependencies neither
